### DISCUSSION TITLE
Putting anchors into a table

### DISCUSSION THREAD

**Discussion by lozdown** [sqlpage/SQLPage#538] [Q&A]
# Background
I populate a table component with rows from a database. Each row is identified using unique column id. Using markdown a column in each row provides a link to another sqlpage page with id as a parameter. This second page updates the database and returns back to the table. I want to return to the row used in the call identified by id. I intended to use an html anchor  but I can't make it work. 

There seems to be two problems 

1. the example url _http://127.0.0.1:8080/member-event.sql#id441963_ does not move to the requested row in the table created by member-event.sql
2. the returning redirect component does not create the correct url 

# Table component definition (cut down) from member-event.sql

```
 select 
 	'table' as 'component', true as 'small',true as striped_columns,
 	TRUE as search,
 	'2024-01-17' as markdown
 	;
 
select m."First Name",m."Last Name",
 	case when "2024-01-17"  
 		then 
 			format('[](#%s)[&#x2713;](remove-member-event.sql?memberid=%s&eventid=%s)', a.contactid,a.contactid,(select distinct id from eventinfo where eventdate='2024-01-17') ) 
		else 
 			format('[](#%s)[x](update-member-event.sql?memberid=%s&eventid=%s)', a.contactid,a.contactid,(select distinct id from eventinfo where eventdate='2024-01-17') )  
		end as "2024-01-17" 
 	from members m left join attendances a on m."Contact ID"=a.contactid
;
 
```
The update-member-event.sql file definition is 
```
INSERT INTO public.events_members(
	events_id, "members_Contact ID")
	VALUES ($eventid::int, $memberid::int)
RETURNING
    'redirect' AS component,
    'member-event.sql#$memberid' AS link
;
```

I was expecting the redirect to return a url something like _member-event.sql#441963_  but it actually returns verbatim (ie  with no substitution of the parameter value) _member-event.sql#$memberid_

**Comment by lovasoa** [comment]
You probably want to do a concatenation instead: 

```sql
CONCAT('member-event.sql#', $memberid)
```

or 

```sql
'member-event.sql#' || $memberid
```

or

```sql
format('member-event.sql#%s', $memberid)
```

**Reply by lovasoa** [reply]
there is unfortunately no id on table rows in the table component currently.

A pull request to introduce that would be welcome!

**Comment by lozdown** [comment]
Thank you for such a fast helpful reply. 

> A pull request to introduce that would be welcome!

I don't actually know how to do this. I will try to find out but it's likely to take me a while.

**Reply by lovasoa** [reply]
I did it today, it was a small change: https://github.com/lovasoa/SQLpage/commit/f46ccf592c70d0620f14fad21651c22b72d6b6f2

it will be in the next version to be published soon

**Comment by lozdown** [comment]
Perfect - thank you

**Comment by lozdown** [comment]
@lovasoa - I tried the updated sqlpage table component software but the behaviour is unexpected. 

My sqlpage file (example: ...member-event.sql) renders the  table correctly with rows correctly identified with the _sqlpage_id parameter. 

```
 .... html extract ....

 <tr class=" " id="389374"><td class="First Name align-middle">John</td>........</tr>
 </tbody>
 </table>
.....
```
If I manually append a #rowid to the url (example: ...member-event.sql#416188) the browser scrolls correctly to the target row.

However if the exact same url with appended #rowid is generated using this redirect component, the browser does not scroll. 

`INSERT INTO public.events_members(
	events_id, "members_Contact ID")
	VALUES ($eventid::int, $memberid::int)
RETURNING
    'redirect' AS component,
    CONCAT('member-event.sql#', $memberid) AS link
;`

If I then manually edit the url and refresh it does scroll.

**Reply by lovasoa** [reply]
Can you try and make a minimal reproduction of that ?

**Comment by lozdown** [comment]
I have created a minimal sqlpage site (using postgresql) that has clarified the problem. 
In summary:

If the table component includes the search parameter, then scroll to #id does not work. 
If the same table does not have the search parameter the scroll to #id does work.

The minimal site uses a database with a single table:
`CREATE TABLE IF NOT EXISTS public.words
(
    id integer NOT NULL DEFAULT nextval('words_id_seq'::regclass),
    word text COLLATE pg_catalog."default" NOT NULL,
    CONSTRAINT words_pkey PRIMARY KEY (id)
)
` 
The sql files are here
[sqlpage-root.zip](https://github.com/user-attachments/files/16645474/sqlpage-root.zip)

The sqlpage.bin is the latest version (SQLPage v0.26.0), the table component is the latest available on the website (includes recent  _sqlpage_id )

**Reply by lovasoa** [reply]
Thanks ! I can reproduce it now. I opened an issue here: https://github.com/lovasoa/SQLpage/issues/542
